numpydoc/numpydoc.py

```diff
@@ -203,12 +203,28 @@ def mangle_signature(app, what, name, obj, options, sig, retann):
     if not hasattr(obj, '__doc__'):
         return
     doc = get_doc_object(obj, config={'show_class_members': False})
-    sig = doc['Signature'] or getattr(obj, '__text_signature__', None)
+    sig = (doc['Signature'] or
+           _clean_text_signature(getattr(obj, '__text_signature__', None)))
     if sig:
         sig = re.sub("^[^(]*", "", sig)
         return sig, ''
 
 
+def _clean_text_signature(sig):
+    if sig is None:
+        return None
+    start_pattern = re.compile(r"^[^(]*\(*")
+    start, end = start_pattern.search(sig).span()
+    start_sig = sig[start:end]
+    sig = sig[end:]
+    sig = re.sub(r"\)", "", sig)
```

Comment:
what is this for? It doesn't appear to pertain to any of the test cases below.

Reply:
I'm trying to first make it so 'func(param1, param2, ..., paramN)` -> `param1, param2, ..., paramN`. I suppose I could get rid of the last `re.sub` and just do `sig = sig[end:-1]`, but I wasn't confident if there would be trailing spaces.